guides/hack/56-modules/03-using-internal.md

```diff
@@ -0,0 +1,95 @@
+# Internal and module level visibility
+One goal of using modules is to separate a code unit's public and private interfaces. If a file is part of a module, its toplevel functions, classes, interfaces, traits, enums, and typedefs can be marked `internal`. An internal toplevel symbol cannot be referenced outside of its module. If a symbol is not internal, it is `public`, and it's part of the public API of the module. You can optionally use the `public` keyword on toplevel symbols to denote them as public. 
```

Comment:
```suggestion
One goal of using modules is to separate public and private APIs. If a file is part of a module, its toplevel functions, classes, interfaces, traits, enums, and typedefs can be marked `internal`. An internal toplevel symbol cannot be referenced outside of its module. If a symbol is not internal, it is `public`, and it's part of the public API of the module. You can optionally use the `public` keyword on toplevel symbols to denote them as public. 
```

'Code unit' isn't a concept we use elsewhere in the docs (I think hackc uses the term internally a little bit, but its notion of a code unit will change with decl-based codegen). I'd also avoid 'interface' here to avoid confusion with `interface` declarations.